### PR TITLE
fix for disabled physics with unloadable entity

### PR DESCRIPTION
**Pull request by birarda**
- fixes a bug where physics would be disabled upon entering a domain if a model entity failed to load

### CODE REVIEW

**Comment by birarda** [comment]
#### Test Plan

This is a fix for an easily reproducible bug. 

1. Run a Sandbox and connect to it.
2. Add an entity in front of your avatar with a URL that points to a model that does not exist.
3. Leave the domain or restart your Interface client and return to the domain. Be sure to reconnect close to the entity with the bad URL.

On master/stable, you should notice that physics will never be enabled in that domain. There will be a message in the log that starts with `Physics disabled until entity loads:`. If you are in a domain where there is a ground (like the home set) you will see that your avatar always stays flying.

With this PR, physics should be enabled even with the entity with a bad URL present. The `Physics disabled until entity loads` messages will not repeat in the log. Your avatar should correctly stand on the ground once physics are enabled.

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -916,6 +916,16 @@ bool RenderableModelEntityItem::contains(const glm::vec3& point) const {
     return false;
 }
 
+bool RenderableModelEntityItem::shouldBePhysical() const {
+    // If we have a model, make sure it hasn't failed to download.
+    // If it has, we'll report back that we shouldn't be physical so that physics aren't held waiting for us to be ready.
+    if (_model && _model->didGeometryRequestFail()) {
```
Comment (Atlante45) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Shouldn't that be `!_model || _model->didGeometryRequestFail()`

Comment (birarda) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: This checks if you have a `_model` and the request for said model failed. If you don't have a model yet, we don't know if the request will fail.

Comment (Atlante45) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Yes, but shouldn't `shouldBePhysical` return `false` if we haven't yet made the request?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5d3e125a0668d57a34bae8f35fb83303d6a2b21a available at: https://gist.github.com/41614e8f6c9754f4fc94c2cf0e576931<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8353-5d3e125.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8353-5d3e125.dmg

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp (hunk 1 above)
Comment (birarda) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Nope - we hold the enabling of physics in the domain based on this - it's supposed to be physical as far as we know at that point.

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: SUCCESS
**Additional Test Results + Observations**: Ran a Sandbox and connected to it and 
added an entity in front of my avatar with a URL that points to a model that does not exist as I called it ```hddff.fbx```.  Visited another domain and returned to the domain and reconnected close to the entity with the bad URL called ```hddff.fbx```.  Physics was enabled with the bad URL present. The Physics disabled until entity loads messages did not appear in the log and avatar correctly stood on the ground.

**Pull request merged**